be/src/vec/functions/function.h

```diff
@@ -299,6 +299,10 @@ class FunctionBuilderImpl : public IFunctionBuilder {
 
     ColumnNumbers get_arguments_that_are_always_constant() const override { return {}; }
 
+    // if a function's get_variadic_argument_types() not override and get_return_type_impl()
+    // result is not compile time be sure, the function should override return true
+    virtual bool dont_append_return_type_name_when_register_function() { return false; }
```

Comment:
impl it in `IFunctionBase`, so we can do check in function factory.
if not possible, add comment.

Reply:
the `register_function()` arg `Creator& ptr` is a FunctionBuilder so cant impl it in IFunctionBase it move to IFunctionBuilder